src/care/host_device_map.h

```diff
@@ -69,6 +70,9 @@ namespace care {
         // emplace a key value pair
         inline void emplace(key_type key, mapped_type val) const {
            m_map->emplace(key, val);
+           if (m_map->size() > m_max_size) {
```

Comment:
Could this check be guarded with a macro (e.g. CARE_DEBUG or CARE_ENABLE_BOUNDS_CHECKING)?

Reply:
I'm open to whatever you guys prefer.